controlpanel/frontend/views/datasource.py

```diff
@@ -213,13 +213,28 @@ def get_success_url(self):
 
 
 class GrantAccess(LoginRequiredMixin, PermissionRequiredMixin, CreateView):
+    context_object_name = 'users3bucket'
     form_class = GrantAccessForm
     model = UserS3Bucket
     permission_required = 'api.create_users3bucket'
+    template_name = 'datasource-access-grant.html'
 
     def get_form_kwargs(self):
         return FormMixin.get_form_kwargs(self)
 
+    def get_context_data(self):
+        context = super().get_context_data()
+        bucket = get_object_or_404(S3Bucket, pk=self.kwargs['pk'])
+        context['bucket'] = bucket
+        access_group = bucket.users3buckets.all().select_related('user')
+        member_ids = [member.user.auth0_id for member in access_group]
+        context['access_group'] = access_group
+        context['users_options'] = User.objects.exclude(
+            auth0_id__isnull=True,
```

Comment:
can `User.auth0_id` even be null? I thought it was the primary key for that table?

Reply:
It can and is - the `concourse` user has no `auth0_id`. This code was added a while back to prevent users accidentally adding the `concourse` user to their data access group. `auth0_id` is the primary key, but it is not prevented from being `null` - an old issue. I think `auth0_id` should not be the primary key, as having more than one user that is not logging in via Auth0 may be useful, but probably YAGNI

Reply:
Interesting 👌